src/Compilers/CSharp/Test/Emit2/Semantics/RefReadonlyParameterTests.cs

```diff
@@ -4720,4 +4799,1022 @@ public class C
+            //     C.Y2 = C.X1;
+            Diagnostic(ErrorCode.ERR_NoImplicitConvCast, "C.X1").WithArguments("delegate*<ref int, void>", "delegate*<in int, void>").WithLocation(13, 12),
+            // (14,12): error CS0266: Cannot implicitly convert type 'delegate*<in int, void>' to 'delegate*<ref int, void>'. An explicit conversion exists (are you missing a cast?)
+            //     C.X2 = C.Y1;
+            Diagnostic(ErrorCode.ERR_NoImplicitConvCast, "C.Y1").WithArguments("delegate*<in int, void>", "delegate*<ref int, void>").WithLocation(14, 12));
+
+        var expectedDiagnostics = new[]
+        {
+            // (13,12): warning CS9510: Reference kind modifier of parameter 'ref int' doesn't match the corresponding parameter 'in int' in target.
+            //     C.Y2 = C.X1;
+            Diagnostic(ErrorCode.WRN_TargetDifferentRefness, "C.X1").WithArguments("ref int", "in int").WithLocation(13, 12),
+            // (14,12): warning CS9510: Reference kind modifier of parameter 'in int' doesn't match the corresponding parameter 'ref int' in target.
+            //     C.X2 = C.Y1;
+            Diagnostic(ErrorCode.WRN_TargetDifferentRefness, "C.Y1").WithArguments("in int", "ref int").WithLocation(14, 12)
+        };
+
+        CompileAndVerify(source2, new[] { comp1Ref }, verify: Verification.Fails, parseOptions: TestOptions.RegularNext, options: TestOptions.UnsafeDebugExe,
+            expectedOutput: "X2 Y2 X1 Y1 Y1 X1").VerifyDiagnostics(expectedDiagnostics);
+        CompileAndVerify(source2, new[] { comp1Ref }, verify: Verification.Fails, options: TestOptions.UnsafeDebugExe,
+            expectedOutput: "X2 Y2 X1 Y1 Y1 X1").VerifyDiagnostics(expectedDiagnostics);
+    }
+
+    [Fact]
+    public void Conversion_FunctionPointer_MethodGroup()
+    {
+        var source = """
+            unsafe
+            {
+                delegate*<int, void> v = &V;
+                delegate*<ref readonly int, void> rr = &RR;
+                delegate*<in int, void> i = &I;
+                delegate*<ref int, void> r = &R;
+                delegate*<out int, void> o = &O;
+
+                rr = &V;
+                rr = &R;
+                rr = &I;
+                rr = &O;
+                v = &RR;
+                i = &RR;
+                r = &RR;
+                o = &RR;
+                r = &I;
+                i = &R;
+
+                static void V(int p) => throw null;
+                static void RR(ref readonly int p) => throw null;
+                static void I(in int p) => throw null;
+                static void R(ref int p) => throw null;
+                static void O(out int p) => throw null;
+            }
+            """;
+        CreateCompilation(new[] { source, RequiresLocationAttributeDefinition }, options: TestOptions.UnsafeDebugExe).VerifyDiagnostics(
```

Comment:
>CreateCompilation [](http://example.com/codeflow?start=8&length=17)

Success cases should be executed

Reply:
Representative sets are executed in tests below, I think.